libs/langchain-openai/src/chat_models.ts

```diff
@@ -730,14 +730,21 @@ export class ChatOpenAI<
           );
         }
         if (openAIMessage.additional_kwargs.function_call?.arguments) {
-          count += await this.getNumTokens(
-            // Remove newlines and spaces
-            JSON.stringify(
-              JSON.parse(
-                openAIMessage.additional_kwargs.function_call?.arguments
+          try {
+            count += await this.getNumTokens(
+              // Remove newlines and spaces
+              JSON.stringify(
+                JSON.parse(
+                  openAIMessage.additional_kwargs.function_call?.arguments
+                )
               )
-            )
-          );
+            );
+          } catch (error) {
+            console.error("Error parsing function arguments", error, JSON.stringify(openAIMessage.additional_kwargs.function_call));
```

Comment:
Your call how you want to approach this kind of logging. I've had this as a patch-package patch for a while. Logging this is useful bc then I can easily see when arguments is invalid JSON

Reply:
Yes, this seems reasonable, thank you!